clientconfig/requests_test.go

```diff
@@ -24,85 +24,126 @@ const (
-	cloudsYamlPath := fmt.Sprintf(cloudsPath, utils.RandomString(10, "clouds"))
-	cloudsPublicYamlPath := fmt.Sprintf(cloudsPath, utils.RandomString(15, "clouds-public"))
-	cloudsSecureYamlPath := fmt.Sprintf(cloudsPath, utils.RandomString(10, "secure"))
+func TestGetCloudFromSecure(t *testing.T) {
+	cloudsPath, _, securePath := getCloudPathsSetEnvVars()
 
-	_ = os.Setenv("OS_CLIENT_CONFIG_FILE", cloudsYamlPath)
-	_ = os.Setenv("OS_CLIENT_VENDOR_FILE", cloudsPublicYamlPath)
-	_ = os.Setenv("OS_CLIENT_Secure_FILE", cloudsSecureYamlPath)
-	_ = os.Setenv("OS_CLOUD", "test")
+	cloudsTemplate := cloudsYamlTemplate
+	cloudsSecureTemplate := cloudsSecureYamlTemplate
+
+	f, err := os.Create(cloudsPath)
+	require.NoError(t, err)
+
+	defer f.Close()
+
+	_, err = f.Write([]byte(cloudsTemplate))
+	require.NoError(t, err)
+
+	f, err = os.Create(securePath)
+	require.NoError(t, err)
+	_, err = f.Write([]byte(cloudsSecureTemplate))
+	require.NoError(t, err)
+
+	cl, err := GetCloudFromYAML(&ClientOpts{})
+	require.NoError(t, err)
+	require.Contains(t, cl.AuthInfo.AuthURL, osAuthUrl)
+	require.Contains(t, cl.AuthInfo.Password, osPassword2)
+}
+
+func TestGetCloudFromAllClouds(t *testing.T) {
+	cloudsPath, publicPath, securePath := getCloudPathsSetEnvVars()
 
 	cloudsTemplate := cloudsYamlTemplate
 	cloudsPublicTemplate := cloudsPublicYamlTemplate
 	cloudsSecureTemplate := cloudsSecureYamlTemplate
 
-	f, err := os.Create(cloudsYamlPath)
+	f, err := os.Create(cloudsPath)
 	require.NoError(t, err)
 
 	defer f.Close()
 
 	_, err = f.Write([]byte(cloudsTemplate))
 	require.NoError(t, err)
 
-	f, err = os.Create(cloudsPublicYamlPath)
+	f, err = os.Create(publicPath)
 	require.NoError(t, err)
 	_, err = f.Write([]byte(cloudsPublicTemplate))
 	require.NoError(t, err)
 
-	f, err = os.Create(cloudsSecureYamlPath)
+	f, err = os.Create(securePath)
 	require.NoError(t, err)
 	_, err = f.Write([]byte(cloudsSecureTemplate))
 	require.NoError(t, err)
 
 	cl, err := GetCloudFromYAML(&ClientOpts{})
 	require.NoError(t, err)
-	require.Contains(t, cl.AuthInfo.AuthURL, "http://url-from-clouds-public.yaml")
-	require.Contains(t, cl.AuthInfo.Password, "SecuredPa$$w0rd1")
+	require.Contains(t, cl.AuthInfo.AuthURL, osAuthUrl2)
+	require.Contains(t, cl.AuthInfo.Password, osPassword2)
+}
+
+func TestGetPureCloud(t *testing.T) {
+	cloudsPath, _, _ := getCloudPathsSetEnvVars()
+	cloudsTemplate := cloudsYamlTemplate
+
+	f, err := os.Create(cloudsPath)
+	require.NoError(t, err)
+
+	defer f.Close()
+
+	_, err = f.Write([]byte(cloudsTemplate))
+	require.NoError(t, err)
+
+	cl, err := GetCloudFromYAML(&ClientOpts{})
+	require.NoError(t, err)
+	require.Contains(t, cl.AuthInfo.AuthURL, osAuthUrl)
+	require.Contains(t, cl.AuthInfo.Password, osPassword)
+	require.Contains(t, cl.AuthInfo.Username, osUsername)
+	require.Contains(t, cl.AuthInfo.ProjectName, osProjectName)
+	require.Contains(t, cl.AuthInfo.UserDomainName, osDomainName)
+}
+
+func getCloudPathsSetEnvVars() (string, string, string) {
```

Comment:
Maybe name it smth like `prepareCloudPaths`? `get...` sound like you're getting "CloudPathsSetEnvVars" from my PoV